server/sonar-server/src/main/java/org/sonar/server/project/ws/GhostsAction.java

```diff
@@ -32,61 +32,83 @@
+import org.sonar.db.organization.OrganizationDto;
 import org.sonar.server.es.SearchOptions;
+import org.sonar.server.organization.DefaultOrganizationProvider;
 import org.sonar.server.user.UserSession;
 
 import static com.google.common.collect.Sets.newHashSet;
 import static org.sonar.server.es.SearchOptions.MAX_LIMIT;
+import static org.sonar.server.ws.WsUtils.checkFoundWithOptional;
 
 public class GhostsAction implements ProjectsWsAction {
-  public static final String ACTION = "ghosts";
+  private static final String PARAM_ORGANIZATION = "organization";
+  private static final String ACTION = "ghosts";
   private static final Set<String> POSSIBLE_FIELDS = newHashSet("uuid", "key", "name", "creationDate");
 
   private final DbClient dbClient;
   private final UserSession userSession;
+  private final DefaultOrganizationProvider defaultOrganizationProvider;
 
-  public GhostsAction(DbClient dbClient, UserSession userSession) {
+  public GhostsAction(DbClient dbClient, UserSession userSession, DefaultOrganizationProvider defaultOrganizationProvider) {
     this.dbClient = dbClient;
     this.userSession = userSession;
+    this.defaultOrganizationProvider = defaultOrganizationProvider;
   }
 
   @Override
   public void define(WebService.NewController context) {
-    context
-      .createAction(ACTION)
+    WebService.NewAction action = context.createAction(ACTION);
+
+    action
       .setDescription("List ghost projects.<br /> Requires 'Administer System' permission.")
       .setResponseExample(Resources.getResource(getClass(), "projects-example-ghosts.json"))
       .setSince("5.2")
       .addPagingParams(100, MAX_LIMIT)
       .addFieldsParam(POSSIBLE_FIELDS)
       .addSearchQuery("sonar", "names", "keys")
       .setHandler(this);
+
+    action.createParam(PARAM_ORGANIZATION)
+      .setDescription("the organization key")
+      .setRequired(false)
+      .setInternal(true)
+      .setSince("6.3");
   }
 
   @Override
   public void handle(Request request, Response response) throws Exception {
-    userSession.checkPermission(UserRole.ADMIN);
-    DbSession dbSession = dbClient.openSession(false);
+    userSession.checkLoggedIn();
+
     SearchOptions searchOptions = new SearchOptions()
       .setPage(request.mandatoryParamAsInt(Param.PAGE),
         request.mandatoryParamAsInt(Param.PAGE_SIZE));
     Set<String> desiredFields = fieldsToReturn(request.paramAsStrings(Param.FIELDS));
     String query = request.param(Param.TEXT_QUERY);
 
-    try {
-      long nbOfProjects = dbClient.componentDao().countGhostProjects(dbSession, query);
-      List<ComponentDto> projects = dbClient.componentDao().selectGhostProjects(dbSession, searchOptions.getOffset(), searchOptions.getLimit(), query);
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      OrganizationDto organization = getOrganization(dbSession, request);
+      userSession.checkOrganizationPermission(organization.getUuid(), UserRole.ADMIN);
```

Comment:
Constants of global permissions are in `GlobalPermissions`. `UserRole`, as it name does not describe..., is for project permissions. So you should replace `UserRole.ADMIN` by `GlobalPermissions.SYSTEM_ADMIN`.